corehq/apps/app_manager/views/utils.py

```diff
@@ -63,56 +63,40 @@ def back_to_main(request, domain, app_id=None, module_id=None, form_id=None,
     which then redirect to the main page.
 
     """
-    page = None
-    params = {}
-    args = [domain]
-    view_name = 'dashboard_default'
+    args = [domain, app_id]
+    view_name = 'view_app'
 
-    form_view = 'form_source'
+    app = get_app(domain, app_id)
 
-    if app_id is not None:
-        view_name = 'view_app'
-        args.append(app_id)
-
-        app = get_app(domain, app_id)
-
-        module = None
+    module = None
+    try:
+        if module_id is not None:
+            module = app.get_module(module_id)
+        elif module_unique_id is not None:
+            module = app.get_module_by_unique_id(module_unique_id)
+    except ModuleNotFoundException:
+        raise Http404()
+
+    form = None
+    if form_id is not None and module is not None:
+        try:
+            form = module.get_form(form_id)
+        except IndexError:
+            raise Http404()
+    elif form_unique_id is not None:
         try:
-            if module_id is not None:
-                module = app.get_module(module_id)
-            elif module_unique_id is not None:
-                module = app.get_module_by_unique_id(module_unique_id)
-        except ModuleNotFoundException:
+            form = app.get_form(form_unique_id)
+        except FormNotFoundException:
             raise Http404()
 
-        form = None
-        if form_id is not None and module is not None:
-            try:
-                form = module.get_form(form_id)
-            except IndexError:
-                raise Http404()
-        elif form_unique_id is not None:
-            try:
-                form = app.get_form(form_unique_id)
-            except FormNotFoundException:
-                raise Http404()
-
-        if form is not None:
-            view_name = 'view_form' if form.no_vellum else form_view
-            args.append(form.unique_id)
-        elif module is not None:
-            view_name = 'view_module'
-            args.append(module.unique_id)
-
-    if page:
-        view_name = page
-
-    return HttpResponseRedirect(
-        "%s%s" % (
-            reverse(view_name, args=args),
-            "?%s" % urlencode(params) if params else ""
-        )
-    )
+    if form is not None:
+        view_name = 'view_form' if form.no_vellum else 'form_source'
```

Comment:
woah! how did that get into the code base, that was an incorrect indent right?

Reply:
Hmm? It's not an incorrect indent, I'm just removing the `if app` checks and unindenting. But this diff might have come out weird.